File: src/one.rs
use crate::utils::read_lines;

pub fn run() {
    if let Ok(lines) = read_lines("inputs/1.txt") {
        let mut calories: Vec<i32> = Vec::new();
        let mut elves: Vec<Vec<i32>> = Vec::new();
        for line in lines {
            if let Ok(ip) = line {
                if ip.is_empty() {
                    elves.push(calories);
                    calories = Vec::new();
                } else {
                    calories.push(ip.parse().unwrap());
                }
            }
        }

        if calories.len() != 0 {
            elves.push(calories);
        }
        let mut sums: Vec<i32> = Vec::new();
        for elf in elves.iter() {
            sums.push(elf.iter().sum());
        }

        println!("{}", sums.iter().max().expect("No Maximum Found"));


        sums.sort();
        println!("{}", sums.iter().rev().take(3).sum::<i32>());
    }
}

File: src/one.exs
path = Path.join(["..", "inputs", "1.txt"])
{:ok, data} = File.read(path)
lines = String.split(data, "\n") |> Enum.chunk_by(&(&1 == ""))
filtered = Enum.map(lines, &(Enum.filter(&1, fn l -> l != "" end)))
ints = Enum.map(filtered, fn list -> Enum.map(list, &(String.to_integer(&1))) end)
totals = Enum.map(ints, &(Enum.sum(&1)))

totals |> Enum.max |> Integer.to_string |> IO.puts
totals |> Enum.sort(&(&1 >= &2)) |> Enum.take(3) |> Enum.sum |> Integer.to_string |> IO.puts
